R/checkers.R

```diff
@@ -45,3 +56,209 @@ check_elements_non_neg <- function(x, arg = "x", call = rlang::caller_env()) {
   }
   invisible()
 }
+
+
+
+#' Check that the arguments are either integers or characters
+#'
+#' @description
+#' This is for unique identifiers of groupings, which we will allow to
+#' either be character or integers.
+#'
+#'
+#' @param x Object with type checking
+#' @param arg Name of the argument supplying the object
+#' @param call Calling environment to be passed to [cli::cli_abort()]
+#'
+#' @return NULL, invisibly
+check_int_or_char <- function(x, arg = "x", call = rlang::caller_env()) {
+  # We only want vectors not lists
+  if (!(rlang::is_integerish(x) || rlang::is_character(x))) {
+    throw_type_error(
+      object = x,
+      arg_name = arg,
+      expected_type = "integer or character",
+      call = call
+    )
+  }
+  invisible()
+}
```

Comment:
This should be done with checkmate functions as well. As below, can either combine assertions like this:

`checkmate::assert(checkmate::check_integerish(x), checkmate::check_character(x))`

or can use `checkmate::check_*()` output in `if` / custom `cli::cli_abort()` calls.